src/test/java/com/google/jenkins/plugins/k8sengine/KubernetesEnginePublisherTest.java

```diff
@@ -360,6 +375,102 @@ public void testDoCheckZoneOKWithMatchingZones() {
     assertEquals(FormValidation.ok(), result);
   }
 
+  @Test
+  public void testDoFillClusterNameItemsEmptyWithEmptyCredentialsId() {
+    testDoFillClusterNameItems(
```

Comment:
Avoiding duplication of code is generally the right approach, however some leeway is generally given in unit tests when it improves readability of tests. The extensive use of helper methods in these tests requires the reader to do a lot of jumping around to understand what/how these tests work. I'll send you a link to a doc which provide some guidance on this topic, and let you decide how best to proceed.